src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -37,6 +37,9 @@ public CppWriter(CppCodegenCompilation compilation, string outputFilePath)
 
             _out = new StreamWriter(new FileStream(outputFilePath, FileMode.Create, FileAccess.Write, FileShare.Read, 4096, false));
 
+            // Emit the Unicode byte order mark to prevent the MS C++ compiler treating the emitted
+            // C++ code as codepage 1251
+            _out.Write('\uFEFF');
```

Comment:
Can you just use the other constructor of `StreamWriter` here? The one that takes an explicit `Encoding` parameter and pass it `Encoding.Utf8` - that one should emit a BOM.

You might also be fixing #663 with this.